### PR TITLE
Сonversion and compression implementation

### PR DESCRIPTION
**Pull request by Konstantsiy**

### CODE REVIEW

File: internal/converter/converter.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	switch targetFormat {
+	case "png":
+		var enc png.Encoder
+		enc.CompressionLevel = png.CompressionLevel(ratio)
+		err := enc.Encode(outputFile, imageData)
+		if err != nil {
+			return nil, err
+		}
+	case "jpeg":
+	case "jpg":
```
Comment (papahmsolo) [inline internal/converter/converter.go:45]: - we can combine these cases `case "jpg", "jpeg"`
- let's create constants for the format values

File: internal/converter/converter.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		{
+			err := jpeg.Encode(outputFile, imageData, &jpeg.Options{
+				Quality: ratio,
+			})
+			if err != nil {
+				return nil, err
+			}
```
Comment (papahmsolo) [inline internal/converter/converter.go:52]: Can we wrap errors?

File: internal/converter/converter.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+// Converter converts and compresses images.
+type Converter struct{}
+
+// NewConverter creates new converter.
+func NewConverter() *Converter {
+	return &Converter{}
+}
+
+//Convert converts and compresses the given image file according to the target format and compression ratio.
+func (c *Converter) Convert(file multipart.File, targetFormat string, ratio int) (*os.File, error) {
```
Comment (papahmsolo) [inline internal/converter/converter.go:25]: Converter is not responsible for storing images, only for converting them. Storing converted image is a part of business logic

**Review by papahmsolo — CHANGES_REQUESTED** [review]
Looks good!

File: internal/converter/converter.go (hunk 3 above)
Comment (Konstantsiy) [reply internal/converter/converter.go:25]: > Converter is not responsible for storing images, only for converting them. Storing converted image is a part of business logic
Converter creates an intermediate file, which is then deleted in the business logic by its filename

File: internal/converter/converter.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+		}
+	case FormatJPEG, FormatJPG:
+		{
+			err := jpeg.Encode(buf, imageData, &jpeg.Options{
+				Quality: ratio,
+			})
+			if err != nil {
+				return nil, fmt.Errorf("can't convert image to %s format: %w", FormatJPG, err)
+			}
+		}
```
Comment (papahmsolo) [inline internal/converter/converter.go:53]: Any reason we want to add additional brackets?

File: internal/converter/converter.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+// Convert converts and compresses the given image file according to the target format and compression ratio.
+func (c *Converter) Convert(file multipart.File, targetFormat string, ratio int) (io.ReadSeeker, error) {
+	imageData, _, err := image.Decode(file)
+	if err != nil {
+		return nil, fmt.Errorf("can't decode source file: %w", err)
+	}
+
+	buf := new(bytes.Buffer)
+
+	switch targetFormat {
```
Comment (papahmsolo) [inline internal/converter/converter.go:37]: What happens if we pass in the format other than png and jpeg?

File: internal/converter/converter.go [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+	FormatPNG  = "png"
+)
+
+// Converter converts and compresses images.
+type Converter struct{}
+
+// NewConverter creates new converter.
+func NewConverter() *Converter {
+	return &Converter{}
+}
```
Comment (papahmsolo) [inline internal/converter/converter.go:26]: Do we need a converter type?

File: internal/converter/converter.go [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+// Convert converts and compresses the given image file according to the target format and compression ratio.
+func (c *Converter) Convert(file multipart.File, targetFormat string, ratio int) (io.ReadSeeker, error) {
```
Comment (papahmsolo) [inline internal/converter/converter.go:29]: `file multipart.File` could be a reader

**Review by papahmsolo — CHANGES_REQUESTED** [review]
👀

File: internal/converter/converter.go (hunk 4 above)
Comment (Konstantsiy) [reply internal/converter/converter.go:53]: > Any reason we want to add additional brackets?

Apparently they were generated automatically, I didn't notice.

File: internal/converter/converter.go (hunk 6 above)
Comment (Konstantsiy) [reply internal/converter/converter.go:26]: > Do we need a converter type?

I agree, I went too far with structuring :)

**Review by papahmsolo — APPROVED** [review]
👍

**Pull request merged**